<launch>
        <arg name="file" />
        <arg name="rings" default="32"/>
	<env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find velodyne_odo)/cfg/rosconsole.config"/>
	 <!--node name="sph_test" pkg="velodyne_odo" type="sph" launch-prefix="optirun" output="screen"-->
	<node name="sph_test" pkg="velodyne_odo" type="sph" output="screen">
	 <!--node name="sph_test" pkg="velodyne_height_map" type="sph_test" output="screen" launch-prefix="valgrind" -->
        <param name="bag_name" value="$(arg file)"/>
		<param name="max_mem_req" value="8"/>
		<param name="wait_input" value="true"/>
		<param name="pause" value="true"/>
		<param name="cadr_tms" value="100"/>
		<param name="cadr_result_tms" value="100"/>
		<param name="key_tms" value="100"/>
		<param name="Xmax" value="100.0"/>
		<param name="Ymax" value="100.0"/>
		<param name="Zmax" value="20.0"/>
		<param name="Zmin" value="-20.0"/>
		<param name="minRad" value="5.0"/>
		<param name="sparse_first" value="0.2"/>
		<param name="sparse_next" value="0.2"/>
		<param name="Xrob" value="0"/>
		<param name="Yrob" value="0"/>
		<param name="Zrob" value="0"/>
		<param name="qx" value="0.0"/>
		<param name="qy" value="0.0"/>
		<param name="qz" value="0.0"/>
		<param name="qw" value="1.0"/>
		<param name="vx0" value="0"/>
		<param name="vy0" value="0"/>
		<param name="vz0" value="0"/>
		<param name="wx0" value="0"/>
		<param name="wy0" value="0"/>
		<param name="wz0" value="0"/>
		<param name="inc" value="1"/>
		<param name="cell_size" value="0.2"/>
		<param name="sph_size" value="1.0"/>
		<param name="start_cloud" value="0"/>
		<param name="finish_cloud" value="99900"/>
		<param name="registration_distance" value="20.0"/>
		<param name="min_dis" value="0.001"/>
		<param name="min_rot" value="0.0001"/>
		<param name="max_dis" value="0.1"/>
		<param name="max_rot" value="0.01"/>
		<param name="max_iterations" value="100"/>
		<param name="max_alignment_iterations" value="100"/>
		<param name="z_map" value="-1.0"/>
		<param name="rays" value="1080"/>
		<param name="rings" value="$(arg rings)"/>
		<param name="correct" value="true"/>
		<param name="active_points_percent" value="0.5"/>
		<param name="min_point_part" value="0.2"/>
		<param name="max_accum_number" value="1" />
	</node>
</launch>
